pallets/asset-manager/src/lib.rs

```diff
@@ -118,24 +115,93 @@ pub mod pallet {
         type WeightInfo: crate::weights::WeightInfo;
     }
 
+    /// Asset Manager Pallet
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    #[pallet::without_storage_info]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    pub struct Pallet<T>(_);
+
+    impl<T> AssetIdType for Pallet<T>
+    where
+        T: Config,
+    {
+        type AssetId = T::AssetId;
+    }
+
+    impl<T> LocationType for Pallet<T>
+    where
+        T: Config,
+    {
+        type Location = T::Location;
+    }
+
+    impl<T> AssetIdLocationMap for Pallet<T>
+    where
+        T: Config,
+    {
+        #[inline]
+        fn location(asset_id: &Self::AssetId) -> Option<Self::Location> {
+            AssetIdLocation::<T>::get(asset_id)
+        }
+
+        #[inline]
+        fn asset_id(location: &Self::Location) -> Option<Self::AssetId> {
+            LocationAssetId::<T>::get(location)
+        }
+    }
+
+    impl<T> assets::UnitsPerSecond for Pallet<T>
+    where
+        T: Config,
+    {
+        #[inline]
+        fn units_per_second(id: &Self::AssetId) -> Option<u128> {
+            UnitsPerSecond::<T>::get(id)
+        }
+    }
+
+    /// Genesis Configuration
     #[pallet::genesis_config]
     pub struct GenesisConfig<T: Config> {
-        pub start_id: AssetId,
-        pub _marker: PhantomData<T>,
+        pub start_id: T::AssetId,
+    }
+
+    #[cfg(feature = "std")]
+    impl<T> GenesisConfig<T>
```

Comment:
why do you provide the `GenesisConfig` type here explicitly if you're just forwarding `build` and `assimilate_storage`?

Reply:
Yeah I don't think it's needed. Removing it unless @bhgomes had something else in mind.